File: projects/poly/code/unmax1.py
# remove the all occurences of the
# largest integer in the list
#   [1,2,3,2,3,1]
# returns:
#   [1,2,2,1]
def unmax(x):
    m = max(x) 

    r = []
    for v in x:
        if v != m:
            r.append(v)

    return r

print(unmax([1,1,2,3,3,4,1,2,7,1]))

File: projects/poly/code/unmax1.go
package main

import "fmt"

// remove the all occurences of the
// largest integer in the list
//   [1,2,3,2,3,1]
// returns:
//   [1,2,2,1]
func unmax(x []int) []int {
	max := 0
	for _, v := range x {
		if v > max {
			max = v
		}
	}

	r := []int{}
	for _, v := range x {
		if v != max {
			r = append(r, v)
		}
	}

	return r
}

func main() {
	fmt.Printf("%v\n", unmax([]int{1, 1, 2, 3, 3, 4, 1, 2, 7, 1}))
}
